app/controllers/api/v1/users/change_password_controller.rb

```diff
@@ -58,7 +60,7 @@ def set_user
                   else
                     token = jsonapi_params[:one_time_token]
                     token_user = User.find_by_one_time_token(token)
-                    raise ActiveRecord::RecordNotFound if token_user.nil?
+                    raise NoSuchResetPasswordToken if token_user.nil?
```

Comment:
Api::V1::Users::ChangePasswordController#set_user performs a nil-check

[Read more about it here.](https://github.com/troessner/reek/blob/master/docs/Nil-Check.md)